electrum/lnpeer.py

```diff
@@ -1790,15 +1790,6 @@ def log_fail_reason(reason: str):
             # TODO fail here if invoice has set PAYMENT_SECRET_REQ
             payment_secret_from_onion = None
 
-        if total_msat > amt_to_forward:
-            mpp_status = self.lnworker.check_received_mpp_htlc(payment_secret_from_onion, chan.short_channel_id, htlc, total_msat)
-            if mpp_status is None:
-                return None, None
-            if mpp_status is False:
-                log_fail_reason(f"MPP_TIMEOUT")
-                raise OnionRoutingFailure(code=OnionFailureCode.MPP_TIMEOUT, data=b'')
-            assert mpp_status is True
```

Comment:
As discussed elsewhere, the next block of code is used for trampoline forwarding -- and this block of code ensures that we wait for all htlcs in an MPP, before we forward trampoline, which we should do. So the ordering is critical here.

Reply:
reverted to previous order